src/sql/SqlService.ts

```diff
@@ -42,6 +42,92 @@ import {HzLocalTime, HzLocalDate, HzLocalDateTime, HzOffsetDateTime} from './Dat
+ * guaranteed between minor or patch releases.
+ *
+ * If this cluster is a Hazelcast Jet cluster, a statement can be executed by either the default SQL backend or by Hazelcast
+ * Jet backend, as a Jet job. If some of the features used in a statement isn't supported by the default backend, the engine will
+ * attempt to execute it using Jet. This class is the API to both backends.
+ *
+ * The text below summarizes features supported by the default SQL engine. For a summary of Hazelcast Jet SQL features
+ * see `com.hazelcast.jet.sql` package javadoc in Hazelcast Jet (once released).
+ *
+ *
+ * ### Overview
+ * Hazelcast is able to execute distributed SQL queries over the following entities:
+ * * IMap
+ *
+ * When an SQL statement is submitted to a member, it is parsed and optimized by the `hazelcast-sql` module, that is based
+ * on <a href="https://calcite.apache.org">Apache Calcite</a>. The `hazelcast-sql` must be in the classpath, otherwise
+ * an exception will be thrown.
+ *
+ * During optimization, a statement is converted into a directed acyclic graph (DAG) that is sent to cluster members
+ * for execution. Results are sent back to the originating member asynchronously and returned to the user via {@link SqlResult}.
+ *
+ * ##### Querying an IMap
+ *
+ * Every IMap instance is exposed as a table with the same name in the `partitioned` schema. The `partitioned`
+ * schema is included into a default search path, therefore an IMap could be referenced in an SQL statement with or without the
+ * schema name.
+ *
+ * ###### Column Resolution
+ * Every table backed by an IMap has a set of columns that are resolved automatically. Column resolution uses IMap entries
+ * located on the member that initiates the query. The engine extracts columns from a key and a value and then merges them
+ * into a single column set. In case the key and the value have columns with the same name, the key takes precedence.
+ *
+ * Columns are extracted from objects as follows:
+ * * For non-Portable objects, public getters and fields are used to populate the column list. For getters, the first
+ *   letter is converted to lower case. A getter takes precedence over a field in case of naming conflict.
+ * * For Portable objects, field names used in the {@link PortableWriter.writePortable} method are used to populate the column
+ * list
+ *
+ * The whole key and value objects could be accessed through a special fields `__key` and `this`, respectively. If
+ * key (value) object has fields, then the whole key (value) field is exposed as a normal field. Otherwise the field is hidden.
+ * Hidden fields can be accessed directly, but are not returned by `SELECT * FROM ...` queries.
+ *
+ * If the member that initiates a query doesn't have local entries for the given IMap, the query fails.
+ *
+ * Consider the following key/value model:
+ *
+ * ```ts
+ * class PersonKey {
+ *         personId: number;
```

Comment:
I think we should make this portable and show writePortable method to show the column names and types. That is the only way of setting the schema from the client side. For the python client, I used such samples

```
    Consider the following key/value model: ::

        class PersonKey(Portable):
            def __init__(self, person_id=None, department_id=None):
                self.person_id = person_id
                self.department_id = department_id

            def write_portable(self, writer):
                writer.write_long("person_id", self.person_id)
                writer.write_long("department_id", self.department_id)

            ...

        class Person(Portable):
            def __init__(self, name=None):
                self.name = name

            def write_portable(self, writer):
                writer.write_string("name", self.name)

            ...
```

Reply:
btw is your example missing factory id and class id

Reply:
Should I delete them too?

Reply:
I think they are unrelated and can be omitted. We should just mention that this is a portable class